stock-tracker/src/features/search/redux/searchMiddleware.js

```diff
@@ -0,0 +1,32 @@
+import { setChartLoadingAction } from "../../chart";
+import { setLoadingOverviewAction } from "../../overview";
+import { setLoadingNewsAction } from "../../latestNews";
+import { setLoadingKeyStatsAction } from "../../keyStats";
+import { setLoadingPeersAction } from "../../topPeers";
+import { setSuggestionsAction } from "./actions";
+import { ADD_SYMBOL, ADD_SEARCH_INPUT } from "./actionTypes";
+import {
+  SYMBOL_INPUT,
+  SEARCH_INPUT,
+  SUGGESTIONS
+} from "../../../socket/eventTypes";
+
+export const searchMiddleware = ({
+  socketService
+}) => store => next => action => {
+  const socket = socketService.create();
```

Comment:
This will actually end up creating the WebSocket on the Redux `@@INIT` event, which may not be what you want.
I recommend moving this inside of the `if` blocks